### PR TITLE
[server] Redirect without hitting sorry endpoint.

### LINKED ISSUE #5183: Users not accepting GitLab ToS should not look like application error in Gitpod
@ghuntley noticed that we've got many requests on gitpod.io/sorry which he (validly) thought may be an indicator to a larger problem. 

On the LoadBalancer, I can see that [many of these requests have gitlab.com as referrer](https://console.cloud.google.com/logs/query;query=resource.type%3D%22http_load_balancer%22%0AhttpRequest.requestUrl:%22sorry%22;timeRange=PT12H;summaryFields=httpRequest%252Freferer:false:32:beginning;cursorTimestamp=2021-08-13T07:12:34.237979Z?project=gitpod-191109).
image

In the [server logs I see entries such as](https://console.cloud.google.com/logs/query;cursorTimestamp=2021-08-13T00:41:05.229Z;query=%22server%22%20%22sorry%22%0Atimestamp%3D%222021-08-13T00:07:37.246Z%22%0AinsertId%3D%22uj5g4ylto7z8qp1u%22;timeRange=PT12H?project=gitpod-191109)
```
Error: 403 Forbidden - You (<username>) must accept the Terms of Service in order to perform this action. Please access GitLab from a web browser to accept these terms.
    at Object.create (/app/node_modules/@gitpod/server/dist/src/auth/errors.js:23:30)
    at Object.create (/app/node_modules/@gitpod/server/dist/src/auth/errors.js:48:30)
    at GitLabAuthProvider.<anonymous> (/app/node_modules/@gitpod/server/dist/src/gitlab/gitlab-auth-provider.js:67:61)
    at Generator.throw (<anonymous>)
    at rejected (/app/node_modules/@gitpod/server/dist/src/gitlab/gitlab-auth-provider.js:17:65)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

To avoid future confusion of people who do operations, this error should not do a redirect to "/sorry" but only send a HTTP 403 with the error page and that's it.

### PR DESCRIPTION
**Pull request by laushinka**
Fixes #5183

## Description
When an unauthorized GitLab user logs into Gitpod, we used to handle that by redirecting to /sorry. However, GitLab doesn't clearly show this to the user, and the user doesn't know that they have an incomplete auth process. 
Now we redirect to the frontend and show the error message, so that the user will know that they should complete the auth process on GitLab.

Screenshot 2021-09-07 at 10 41 24

## Related Issue(s)
Fixes #5183

## How to test
Same as how @AlexTugarev described the reproduction steps [here](https://github.com/gitpod-io/gitpod/issues/5183#issuecomment-910213892):

## Release Notes

```
GitLab users who have not completed the auth process on GitLab will be notified with an error message. ([#5568](https://github.com/gitpod-io/gitpod/pull/5568))
```

### CODE REVIEW

**Comment by roboquat** [comment]
@laushinka: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **laushinka** after the PR has been reviewed.

Associated issue: *#5183*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by laushinka** [comment]
@JanKoehnlein This does affect what the user sees, but I'm not sure if this should go into Changelog?

**Comment by JanKoehnlein** [comment]
@laushinka If in doubt, put it into changelog

**Comment by gtsiolis** [comment]
Looking at this now! 👀

**Comment by corneliusludmann** [comment]
Seems not exactly to touch this PR, but anyways: Does it perhaps make sense to make it clearer in the error message that it is the GitLab ToS and not the Gitpod ToS? Could imagine that this is easy to read over. :thinking:

**Comment by laushinka** [comment]
> Seems not exactly to touch this PR, but anyways: Does it perhaps make sense to make it clearer in the error message that it is the GitLab ToS and not the Gitpod ToS? Could imagine that this is easy to read over. 🤔

@corneliusludmann Yes that totally makes sense!

**Review by gtsiolis — APPROVED** [review]
UX looks great, @laushinka! Much better than hitting an error page. 🔮 

However, I agree with what @corneliusludmann mentioned in https://github.com/gitpod-io/gitpod/pull/5568#issuecomment-914352185 about the message clarity. This could also follow the voice and tone used in other error messages we show up there like in https://github.com/gitpod-io/gitpod/pull/3950. ❗ 

Feel free to open a follow up issue or PR for improving that message. 🏓 

🍊 🍊 🍊 🍊 

I'd suggest to rephrase from:

> Error: 403 Forbidden - You (`@username`) must accept the Terms of Service in order to perform this action. Please access GitLab from a web browser to accept these terms.

To:

> The selected provider has updated their **Terms of Service**. Please, visit the provider to review and accept the terms before using **@username** to login in Gitpod.

| BEFORE | AFTER |
|-|-|
| error-before | error-after |

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: c13e2c180fdddd9a8975f66caa208db42cb1fbb6</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5568#pullrequestreview-748074012" title="Approved">gtsiolis</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **laushinka** after the PR has been reviewed.

Associated issue: *#5183*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by laushinka** [comment]
> I'd suggest to rephrase from:
> 
> > Error: 403 Forbidden - You (`@username`) must accept the Terms of Service in order to perform this action. Please access GitLab from a web browser to accept these terms.
> 
> To:
> 
> > The selected provider has updated their **Terms of Service**. Please, visit the provider to review and accept the terms before using **@username** to login in Gitpod.
> 
> BEFORE	AFTER
> error-before	error-after

@gtsiolis Much better text! Will do the change :)

**Comment by gtsiolis** [comment]
> Should we make it specific to GitLab?

Hey @laushinka! In the scope of this PR and #5183, making it specific to GitLab could suffice.

> I think this error only happens with them.

That's correct!

🍋 🍋 🍋 🍋 

However, keep in mind that:
1. This issue occurred due a recent update for the ToS and Privacy Policy for GitLab.com users a few months ago, see https://gitlab.com/gitlab-org/gitlab-foss/-/issues/44798. 💡 
2. ToS and Privacy Policy are also an Admin Area feature which allows GitLab.com admins and GitLab self-hosted admins to define new ToS and Privacy Policy and require new and existing users must accept the terms, see https://docs.gitlab.com/ee/user/admin_area/settings/terms.html. 💡 
3. New changes to ToS and Privacy Policy can be re-enforced for both GitLab.com and self-hosted GitLab instances by design. 💡 

This means that:
1. Both GitLab.com and GitLab self-hosted instances could update their ToS and Privacy Policy and require users to accept the terms. ❗
2. This issue is related to GitLab.com and GitLab self-hosted instances, however, including the term GitLab in the error error message could be confusing if a user selects a GitLab self-hosted instance to login. ❗

**Comment by laushinka** [comment]
> including the term GitLab in the error message could be confusing if a user selects a GitLab self-hosted instance to login.

@gtsiolis Ahh, that makes sense.

**Comment by laushinka** [comment]
Wait I actually can't find where this string comes from. @corneliusludmann do you know where the 403 error message comes from? Is it a response from GitLab?

**Comment by laushinka** [comment]
@gtsiolis @corneliusludmann I think this message isn't under our control unfortunately, or at least I couldn't find it.. So maybe it's a good idea to have this as a separate issue after all.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5568#pullrequestreview-748074012" title="Approved">gtsiolis</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/5568#issuecomment-915205553" title="LGTM">JanKoehnlein</a>*

Associated issue: *#5183*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [JanKoehnlein]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
